packages/ui/react-ui-editor/src/extensions/cursor-line-margin.ts

```diff
@@ -5,34 +5,33 @@
 // Based upon @acheronfail’s implementation, fetched 16 Feb 2024
 // https://discuss.codemirror.net/t/cursorscrollmargin-for-v6/7448/5
 
-import { type EditorState, type Extension, Facet, Transaction } from '@codemirror/state';
+import { type EditorState, type Extension, Transaction } from '@codemirror/state';
 import { EditorView } from '@codemirror/view';
 
 import { log } from '@dxos/log';
 
-/**
- * Number of lines above/below the cursor to keep visible in the viewport.
- * Defaults to 3.
- */
-export const cursorLineMarginFacet = Facet.define<number, number>({
-  combine: (input) => input[0] ?? 3,
-});
-
 const annotation = 'cursorLineMargin';
 const lineAtPos = (s: EditorState, pos: number) => s.doc.lineAt(pos).number;
 
-// Seems to the best approximation of CM5's `cursorScrollMargin`
-// https://discuss.codemirror.net/t/cursorscrollmargin-for-v6/7448
-export const cursorLineMargin: Extension = EditorView.updateListener.of(
-  ({ transactions, state, selectionSet, startState, view }) => {
+type Options = {
+  lines?: number;
+};
+
+// TODO(burdon): Remove. Buggy and creates scrolling problems. Find better (more general) way to deal with HALO notch.
```

Comment:
gets stuck or just disappears?

to me the app felt really broken. clicking would scroll the window. worse, cursoring up forces the window to scroll. i think it's rare that the notch will cover the text (small screen, no sidebar), and it's uneven to have this only work for the text editor.

Reply:
> clicking would scroll the window
If by ‘window’ you mean CM’s overflow parent, that is a good thing if moving the cursor to the beginning of the line would otherwise get it occluded by the notch

> cursoring up forces the window to scroll
This is definitely a bug that should be addressed, but IMO not an indication that the solution has no value. At least the cursor remains visible.

> it's uneven to have this only work for the text editor
I’m not sure yet, but because this uses `scrollIntoView` I think this can work, or already does work, for text editors in stacks and threads.

> i think it's rare that the notch will cover the text (small screen, no sidebar)
This is a fair point, perhaps this behavior only activates when the notch might possibly occlude the cursor, otherwise it is disabled.